cloudigrade/account/migrations/0007_add_image_properties.py

```diff
@@ -0,0 +1,73 @@
+# Generated by Django 2.0.5 on 2018-06-01 10:14
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def migrate_is_windows_to_property(apps, schema_editor):
+    """Copy old is_windows value to an image tag."""
+    ContentType = apps.get_model('contenttypes', 'ContentType')
+
+    # Create image tags
+    ImageTag = apps.get_model('account', 'ImageTag')
+    image_tag_ct = ContentType.objects.get_for_model(ImageTag)
+
+    windows_tag = ImageTag.objects.create(
+        description='windows', polymorphic_ctype=image_tag_ct)
+    ImageTag.objects.create(description='rhel', polymorphic_ctype=image_tag_ct)
```

Comment:
I don't think we need the polymorphic stuff on the `ImageTag` class; I'm struggling to think of when it might be sub-classed since the data we're putting there should be cloud-agnostic.

You're getting the `PolymorphicModel` inheritance chain here that requires that extra kwarg because `ImageTag` is extending `BaseModel`. I think it would be a good idea now to create perhaps _two different_ "base" model classes: one that is and one that isn't polymorphic. In `models.py` they would basically have the exact same definition except for their name and the classes they extend.

Here's a breakdown of what I'm thinking we'd change, and I think it should be an easy clean shuffle:

- Rename `BaseModel` to `BasePolymorphicModel`.
- Change `Account`, `Instance`, `InstanceEvent`, and `MachineImage` to extend `BasePolymorphicModel`.
- Create a new `BaseModel` that simply extends `django.db.models.Model` and replicates the `Meta`, `created_at`, and `updated_at` bits from `BasePolymorphicModel`. (Could share a mixin? I'm not sure about that, though, because using mixins on Django models to change `Meta` can do unexpected things. It might just have to be a little bit of duplicated code.)
- Change `ImageTag` to extend `BaseModel`.
- Remove the unnecessary `polymorphic_ctype`s on the `ImageTag` object creations.

Reply:
(This is of course assuming that being polymorphic doesn't _become_ necessary as some side-effect of interacting with the other polymorphic `MachineImage` class. I confess I haven't actually tried this myself.)

Reply:
I will give it a try and let everyone know!